tests/phpunit/CRM/Contact/BAO/GroupTest.php

```diff
@@ -73,6 +73,52 @@ public function testAddSimple() {
   }
 
   /**
+   * Test case to ensure child group is present in the hierarchy
+   *  if it has multiple parent groups and not all are disabled.
+   */
+  public function testGroupHirerchy() {
+    // Use-case :
+    // 1. Create two parent group A and B and disable B
+    // 2. Create a child group C
+    // 3. Ensure that Group C is present in the group hierarchy
+    $params = array(
+      'name' => uniqid(),
+      'title' => 'Parent Group A',
+      'description' => 'Parent Group One',
+      'visibility' => 'User and User Admin Only',
+      'is_active' => 1,
+    );
+    $group1 = CRM_Contact_BAO_Group::create($params);
+
+    $params = array_merge($params, array(
+      'name' => uniqid(),
+      'title' => 'Parent Group B',
+      'description' => 'Parent Group Two',
+      'is_active' => 0, // disable
+    ));
+    $group2 = CRM_Contact_BAO_Group::create($params);
+
+    $params = array_merge($params, array(
+      'name' => uniqid(),
+      'title' => 'Child Group C',
+      'description' => 'Child Group C',
+      'parents' => array(
+        $group1->id => 1,
+        $group2->id => 1,
+      ),
+    ));
+    $group3 = CRM_Contact_BAO_Group::create($params);
+
+    $params = array(
+      $group1->id => 1,
+      $group3->id => 1,
+    );
+    $groupsHierarchy = CRM_Contact_BAO_Group::getGroupsHierarchy($params, NULL, '&nbsp;&nbsp;', TRUE);
+    // check if child group is present in the tree with formatted group title prepended with spacer '&nbsp;&nbsp;'
+    $this->assertEquals('&nbsp;&nbsp;Child Group C', $groupsHierarchy[$group3->id]);
+  }
+
+  /**
```

Comment:
Be nice if it could check both ways too, so we don't break it in future - ie. disable A.